app/src/main/java/com/inveitix/android/clue/adapters/RecListAdapter.java

```diff
@@ -0,0 +1,123 @@
+package com.inveitix.android.clue.adapters;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.inveitix.android.clue.R;
+import com.inveitix.android.clue.cmn.Museum;
+import com.inveitix.android.clue.interfaces.RecyclerViewOnItemClickListener;
+
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+
+
+public class RecListAdapter extends RecyclerView.Adapter<RecListAdapter.ViewHolder> {
+    public static final int DOWNLOAD_FINISHED = 1;
+    public static final int DOWNLOADING = 2;
+
+
+    private final OnDownloadClickedListener listener;
+    private RecyclerViewOnItemClickListener itemClickListener;
+    private List<Museum> museums;
+    private Context context;
+
+    public RecListAdapter(Context context, List<Museum> museums, OnDownloadClickedListener listener) {
+        this.museums = museums;
+        this.context = context;
+        this.listener = listener;
+    }
+
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_museum, parent, false);
+        ViewHolder vh = new ViewHolder(view);
+        return vh;
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        Museum museum = museums.get(position);
+        holder.txtName.setText(museum.name);
+        holder.museumID = museum.getMuseumID();
+
+        if (museum.getMapStatus() == Museum.DOWNLOADED_STATUS) {
+            holder.btnDownload.setVisibility(View.INVISIBLE);
+            holder.progressBar.setVisibility(View.GONE);
+        } else if (museum.getMapStatus() == Museum.NOT_DOWNLOADED) {
+            holder.btnDownload.setVisibility(View.VISIBLE);
+            holder.progressBar.setVisibility(View.GONE);
+        } else {
+            holder.btnDownload.setVisibility(View.INVISIBLE);
+            holder.progressBar.setVisibility(View.VISIBLE);
+        }
+
+    }
+
+
+    @Override
+    public int getItemCount() {
+        return museums.size();
+    }
+
+    public void setOnItemClickListener(final RecyclerViewOnItemClickListener listener) {
+        this.itemClickListener = listener;
+    }
+
+    public void updateItem(int museumID, int operationId) {
```

Comment:
operationId should be called museumStatus and should pass the constants in Museum